### PR TITLE
fix: correct query filter string generation

### PR DESCRIPTION
**Pull request by noah-eigenfeld**
#### What do these changes do/fix?
Implements: https://github.ibm.com/Watson-Discovery/disco-issue-tracker/issues/9339

Fixes the default logic assembling query filter string anytime that facets are applied to a search.

~This change should add parens ( ) around the values for any fields in the filter param. That way, if there are multiple values for one facet, they are correctly bundled together. Hopefully, it should look something like `facet-name:("value1"|"value2"|"value3"),facet-2-name:("value4"), . . .`.~

The parens I tried to add were getting stripped out by a followup step that gets rid of special characters. Instead, I opted to stack the facet-name/value pairs. They should now look like this:
```
filter: facet-name:"value1"|facet-name:"value2"|facet-name:"value3",facet-2-name:"value4", . . .
```

#### How do you test/verify these changes?
1. Start the example app
2. Perform a search
3. Filter results with the facets panel
4. See that the query request that gets sent out when you change facets has the correctly formatted `filter` string
#### Have you documented your changes (if necessary)?
N/A
#### Are there any breaking changes included in this pull request?
No

### CODE REVIEW

File: packages/discovery-react-components/src/components/SearchFacets/utils/__tests__/searchFilterTransform.test.ts [hunk 1]
```diff
@@ -112,14 +113,20 @@ describe('Filter string to QueryTermAggregation array', () => {
   });
 
   test('it properly handles aggregations with reserved characters', () => {
+    //TODO: failing
+    console.log(
```
Comment (noah-eigenfeld) [inline packages/discovery-react-components/src/components/SearchFacets/utils/__tests__/searchFilterTransform.test.ts:117]: Gotta remove these

File: packages/discovery-react-components/src/components/SearchFacets/utils/searchFilterTransform.tsx [hunk 2]
```diff
@@ -79,7 +83,7 @@ export class SearchFilterTransform {
       const results = get(facet, 'results', []);
       const keys = this.quoteSelectedFacets(results, 'key');
       if (keys.length) {
-        filterStrings.push(`${escapeFieldName(field)}:${keys.join('|')}`);
+        filterStrings.push(keys.map(key => `${escapeFieldName(field)}:${key}`).join('|'));
```
Comment (noah-eigenfeld) [inline packages/discovery-react-components/src/components/SearchFacets/utils/searchFilterTransform.tsx:86]: This is the main functional change for this PR

**Review by broulaye — APPROVED** [review]
Checked it locally, looks good to me 👍

**Review by jhpedemonte — APPROVED** [review]
Tested using example app (tip: had to delete `/examples/discovery-search-app/node_modules` and re-run `yarn` in order to pick up latest changes). In a project with the same file under 2 different names and a facet on `extracted_metadata.filename`, selecting both files in the facet on an empty search showed both results (previously, would only show the first file).

**Pull request merged**